### PR TITLE
Add a draft mock local module and subworkflow

### PR DESCRIPTION
**Pull request by abhi18av**
Sharing the live-coded draft of a module and a subworkflow

### CODE REVIEW

File: modules/local/CAT_pair_unpair.nf [hunk 1]
```diff
@@ -1,32 +1,29 @@
...
-		tuple val(meta), path('*_trimm_pair_R2.fastq.gz')	, emit: p2
-		tuple val(meta), path('*_trimm_unpair_R2.fastq.gz')	, emit: up2
-		
-		output:
-		tuple val(meta), path('*_trimm_cat_R1.fastq.gz')	, emit: c1
-		tuple val(meta), path('*_trimm_cat_R2.fastq.gz')	, emit: c2
-		tuple val(meta), path('*.log')						, emit: log
-		path "versions.yml"									, emit: versions
+        input:
+        tuple val(meta), path('*_trimm_pair_R1.fastq.gz'), path('*_trimm_pair_R2.fastq.gz')
```
Comment (agudeloromero) [inline modules/local/CAT_pair_unpair.nf:8]: In the input is not needed the emmit parameter, this is only for output to say the channel

File: modules/local/CAT_pair_unpair.nf [hunk 2]
```diff
@@ -1,32 +1,29 @@
...
-		tuple val(meta), path('*_trimm_unpair_R2.fastq.gz')	, emit: up2
-		
-		output:
-		tuple val(meta), path('*_trimm_cat_R1.fastq.gz')	, emit: c1
-		tuple val(meta), path('*_trimm_cat_R2.fastq.gz')	, emit: c2
-		tuple val(meta), path('*.log')						, emit: log
-		path "versions.yml"									, emit: versions
+        input:
+        tuple val(meta), path('*_trimm_pair_R1.fastq.gz'), path('*_trimm_pair_R2.fastq.gz')	
+        tuple val(meta), path('*_trimm_unpair_R1.fastq.gz'), path('*_trimm_unpair_R2.fastq.gz')
```
Comment (agudeloromero) [inline modules/local/CAT_pair_unpair.nf:9]: for the channel purpose, it is better combine R1 + R2

File: modules/local/CAT_pair_unpair.nf [hunk 3]
```diff
@@ -1,32 +1,29 @@
...
+        
+        output:
+        tuple val(meta), path('*_trimm_cat_R1.fastq.gz'), path('*_trimm_cat_R2.fastq.gz')	, emit: concatenated
```
Comment (agudeloromero) [inline modules/local/CAT_pair_unpair.nf:12]: Here, I need to use emit and put together pair samples

File: modules/local/CAT_pair_unpair.nf [hunk 4]
```diff
@@ -1,32 +1,29 @@
...
-		"""
-		cat $p1 $up1 > $c1 ;\\
-		cat $p2 $.up2 > $c2 \\
-		> ${prefix}.CAT.pair_unpair.log
+        script:
+    //		def args = task.ext.args ?: '-7'
+        def prefix = task.ext.prefix ?: "${meta.id}"
+        """
+        cat $p1 $up1 > $c1 ;\\
+        cat $p2 $.up2 > $c2 \\
```
Comment (agudeloromero) [inline modules/local/CAT_pair_unpair.nf:21]: I need to remove this dot  $.up2

File: modules/local/bbamp_phix.nf [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+        script:
+        def args = task.ext.args ?: '-Xmx20000m'
+        def args2 = task.ext.args ?: 'ordered cardinality k=31 hdist=1'
+        def prefix = task.ext.prefix ?: "${meta.id}"
+
+        """
+        bbduk.sh \\
+          --cores $task.cpus \\ // do I need this?
+          $args \\
+          in1=$reads[0] in2=$reads[1] \\ // I don't know how to call R1 + R2
```
Comment (agudeloromero) [inline modules/local/bbamp_phix.nf:26]: useful trick, use [0] and [1] to call R1 + R2

File: subworkflows/local/host_removal_pe.nf [hunk 6]
```diff
@@ -0,0 +1,20 @@
+/* https://github.com/agudeloromero/EVEREST/blob/main/SMK/03_Host_removal_PE.smk */
```
Comment (agudeloromero) [inline subworkflows/local/host_removal_pe.nf:1]: Complete subnetwork, this part call the processes

File: workflows/everest.nf [hunk 7]
```diff
@@ -76,15 +76,30 @@ workflow EVEREST {
     //
     // MODULE: Run FastQC
     //
-    FASTQC (
-        INPUT_CHECK.out.reads
-    )
-    ch_versions = ch_versions.mix(FASTQC.out.versions.first())
+    /* FASTQC ( */
```
Comment (agudeloromero) [inline workflows/everest.nf:79]: We will test one part at a time, this is why this is commented

File: workflows/everest.nf [hunk 8]
```diff
@@ -76,15 +76,30 @@ workflow EVEREST {
...
+    /*     INPUT_CHECK.out.reads */
+    /* ) */
+    /* ch_versions = ch_versions.mix(FASTQC.out.versions.first()) */
 
     CUSTOM_DUMPSOFTWAREVERSIONS (
         ch_versions.unique().collectFile(name: 'collated_versions.yml')
     )
 
+    //============================
```
Comment (agudeloromero) [inline workflows/everest.nf:88]: We call the subnetworks here

**Review by agudeloromero — APPROVED** [review]

**Comment by abhi18av** [comment]
Hey @agudeloromero , just to confirm - did you intend to close the PR or merge it? 

At this stage, the code has not been merged 😅

**Pull request closed without merging**